Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -298,32 +350,127 @@ theorem degrees_esymm [Nontrivial R] (n : ℕ) (hpos : 0 < n) (hn : n ≤ Fintyp
 
 end ElementarySymmetric
 
+section CompleteHomogeneousSymmetric
+
+open Finset Multiset Sym
+
+variable [DecidableEq σ] [DecidableEq τ] (n : ℕ)
+
+/-- The `n`th complete homogeneous symmetric `MvPolynomial σ R`. -/
+def hsymm : MvPolynomial σ R := ∑ s : Sym σ n, (s.1.map X).prod
+
+lemma hsymm_def : hsymm σ R n = ∑ s : Sym σ n, (s.1.map X).prod := rfl
+
+/-- `hsymmMu` is the product of the symmetric polynomials `hsymm μᵢ`,
+where `μ = (μ₁, μ₂, ...)` is a partition. -/
+def hsymmMu {n : ℕ} (μ : n.Partition) : MvPolynomial σ R := muProduct σ R (hsymm σ R) μ
+
+lemma hsymmMu_def {n : ℕ} (μ : n.Partition) : hsymmMu σ R μ = (μ.parts.map (hsymm σ R)).prod := rfl
+
+@[simp]
+theorem hsymm_zero : hsymm σ R 0 = 1 := by simp [hsymm, eq_nil_of_card_zero]
+
+@[simp]
+theorem hsymm_one : hsymm σ R 1 = ∑ i, X i := by
+  symm
+  apply Fintype.sum_equiv oneEquiv
+  simp only [oneEquiv_apply, Multiset.map_singleton, Multiset.prod_singleton, implies_true]
+
+theorem hsymmMu_zero : hsymmMu σ R (Nat.Partition.indiscrete 0) = 1 := by simp [hsymmMu]
+
+@[simp]
+theorem hsymmMu_onePart : hsymmMu σ R (Nat.Partition.indiscrete n) = hsymm σ R n := by
+  cases n <;> simp [hsymmMu]
+
+theorem map_hsymm (f : R →+* S) : map f (hsymm σ R n) = hsymm σ S n := by
+  simp [hsymm, ← Multiset.prod_hom']
+
+theorem rename_hsymm (e : σ ≃ τ) : rename e (hsymm σ R n) = hsymm τ R n := by
+  simp_rw [hsymm, map_sum, ← prod_hom', rename_X]
+  apply Fintype.sum_equiv (equivCongr e)
+  simp
+
+theorem hsymm_isSymmetric : IsSymmetric (hsymm σ R n) := rename_hsymm _ _ n
+
+end CompleteHomogeneousSymmetric
+
 section PowerSum
 
 open Finset
 
-variable (σ R) [CommSemiring R] [Fintype σ] [Fintype τ]
+variable (n : ℕ)
```

Comment:
It is (usually) a bad idea to put explicit arguments in `variable` statements as you tend to forget they are there when writing lemma statements. It's better to instead only have the implicit variables declared in a `variable` statement.

Reply:
Yes please :)

Reply:
Done!

Reply:
Of course this screwed up many things.

Reply:
Should I explicitly declare these variables every time (which is a bit cumbersome) or make them implicit and change the rest of the library accordingly? I'm leaning towards the latter, as declaring explicitly `\sigma` and `R` every time is really annoying for someone who's working on symmetric functions. Is that ok?

Reply:
Oh sorry, I see those are used explicitly almost everywhere. I would like to retract my first claim and declare `variable (σ R)` is fine here

Reply:
Uh I already changed it everywhere, should I revert?

Reply:
Yes sorry 🤐

Reply:
Well you said 'explicit variables', not `(n : \N)`. Anyway, reverting is a quick process, I did it!

Reply:
True, but as a matter of fact reviewers shouldn't ask you to change an existing pattern in a PR that adds new code. That's a bad reviewing practice